MyLinkedList.java

```diff
@@ -1 +1,147 @@
+package com.company;
 
+public class MyLinkedList<T> implements List<T> {
+
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+    private StringBuilder stringBuilder = new StringBuilder();
+
+    @Override
+    public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (isEmpty()) {
+            first = newNode;
+            last = newNode;
+        } else {
+            last.next = newNode;
+            newNode.prev = last;
+            last = newNode;
+        }
+        size++;
+    }
+
+    @Override
+    public void add(T value, int index) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (index == 0) {
+            add(value);
+        }
+        if (index == size) {
+            last.next = newNode;
+            last = newNode;
+        }
+        Node<T> leftNode = currentNode(index);
+        Node<T> rightNode = leftNode.prev;
+        leftNode.next = newNode;
+        rightNode.prev = newNode;
+        newNode.prev = leftNode;
+        newNode.next = rightNode;
+        size++;
+    }
+
+    @Override
+    public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
```

Comment:
what happens when we have an empty list?

Reply:
I added if (isEmpty())